code/modules/clothing/suits/armor.dm

```diff
@@ -416,38 +421,43 @@
 	var/repulse_power = 3
 	/// How far away are we finding things to throw
 	var/repulse_range = 5
+	/// What the sparkles looks like
+	var/sparkle_path = /obj/effect/temp_visual/gravpush
 
 /obj/item/clothing/suit/armor/reactive/repulse/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
 	if(!active)
 		return FALSE
 	if(reaction_check(hitby))
 		owner.visible_message("<span class='danger'>[src] blocks [attack_text], converting the attack into a wave of force!</span>")
-		var/list/thrownatoms = list()
+		var/list/thrown_atoms = list()
 		for(var/turf/T in range(repulse_range, owner)) //Done this way so things don't get thrown all around hilariously.
 			for(var/atom/movable/AM in T)
-				thrownatoms += AM
+				thrown_atoms += AM
 
-		for(var/am in thrownatoms)
+		for(var/am in thrown_atoms)
 			var/atom/movable/AM = am
 			if(AM == owner || AM.anchored)
 				continue
 
-			var/throwtarget = get_edge_target_turf(owner, get_dir(owner, get_step_away(AM, owner)))
-			var/distfromuser = get_dist(owner, AM)
-			if(distfromuser == 0)
+			var/throw_target = get_edge_target_turf(owner, get_dir(owner, get_step_away(AM, owner)))
+			var/dist_from_user = get_dist(owner, AM)
+			if(dist_from_user == 0)
 				if(isliving(AM))
 					var/mob/living/M = AM
 					M.Weaken(3)
 					to_chat(M, "<span class='userdanger'>You're slammed into the floor by [owner]'s reactive armor!</span>")
 			else
+				new sparkle_path(get_turf(AM), get_dir(owner, AM))
 				if(isliving(AM))
 					var/mob/living/M = AM
 					to_chat(M, "<span class='userdanger'>You're thrown back by the [owner]'s reactive armor!</span>")
-				spawn(0)
-					AM.throw_at(throwtarget, ((clamp((repulse_power - (clamp(distfromuser - 2, 0, distfromuser))), 3, repulse_power))), 1)//So stuff gets tossed around at the same time.
+				INVOKE_ASYNC(src, .proc/repulse, AM, throw_target, repulse_power, dist_from_user)//So stuff gets tossed around at the same time.
```

Comment:
I'm pretty sure you can skip the repulse proc and just do:
```dm
INVOKE_ASYNC(AM, .proc/throw_at, throw_target, ...) // calculations etc.
```

Reply:
oh yeah that makes a lot more sense